### PR TITLE
Configuration changes: make a few metadata fields repeatable

### LINKED ISSUE #352: Metadata Configuration Changes
Michelle found some scenarios in the map collection that might require us to make some configuration changes so the metadata can be more flexible to handle these scenarios. 

`Barcode` and `Library Catalog Link` fields are currently not repeatable (not multi-value fields), but because of the fact that some map objects are split up into a few objects these fields really do need to be repeatable, as one item (map) might have multiple barcodes/catalog links associated with it. 

From Michelle:

> One that note, this has brought up a metadata issues. Right now, we have the Item Barcode and Library Catalog Link as Not Repeatable, but due to really bizarre cataloging where a split map is sometimes cataloged as two separate items, there might be instances where we do in fact need to repeat these fields. I wasn’t sure if our decisions on repeatability for those fields was hard coded, so I wanted to bring that to your attention, because I think we will have to change that. 

Tasks
- [x] In the Repository Item (islandora_object) content type, update the Barcode field to be repeatable
- [x] In the Repository Item (islandora_object) content type, update the Library Catalog Link field to be repeatable
- [x] In the Repository Item (islandora_object) content type, update the Finding Aid field to be repeatable. 
- [x] In the Collection Object (collection_object) content type, update the Finding Aid field to be repeatable. 
- [x] update the Repository Item's migration yml code to accept multiple values in these fields (add logic to process `|` in the field)
- [x] Update tests in 10-migration-backend-tests.sh:
     - [x] update one of the migrations to have multiple values in these fields. 
     - [x] update the verification code accordingly to check for these multiple values
- [x] Update export view to include changes to these three fields (to output multiple values for them correctly)
- [ ] ~~Update UI to display them correctly~~ - ticketed separately: #365

### PR DESCRIPTION
**Pull request by birkland**
This makes the Finding Aid URLs a multi-value field on both the Repository Items and Collection Object.  It also makes the Item Barcode and Library Catalog Link multi-value fields on the Repository Item.

To test:  load the branch and create a Repository Item.  Give it multiple Finding Aid URL's, Item Barcodes and Library Catalog Links. Ensure that it saves them correctly. 

Try adding more than one Finding Aid on a Collection Object. 

Run the tests, specifically the ones in `10-migration-backend-tests`

Resolves jhu-idc/idc-general#352

### CODE REVIEW

**Comment by bseeger** [comment]
@birkland and @jaredgalanis  - this does affect the Repository Item page as I suspect that doesn't expect these fields to be an array. Should I update those as well or create a separate ticket for @jaredgalanis for this?  This is kind of the first time we've hit this scenario, but I'll keep it in mind that we are now affecting the UI work when we change metadata configuration.

**Comment by bseeger** [comment]
@birkland - this also ripples now into export - so I need to make a few more changes for that to add to this. So, not done yet.

**Comment by birkland** [comment]
@bseeger  I'd say just do it and issue a PR to the UI module if it's straightforward for you.  Otherwise, do create a ticket for Jared.  My understanding, then, is that you'd need to update this PR with a bump to export (for support of newly-multivalued fields), and a bump of the UI module?

**Comment by birkland** [comment]
@bseeger Does this ticket have similar implications to export and UI?  Maybe a more basic question - is the work in that ticket substantially done already?
https://github.com/jhu-idc/iDC-general/issues/305

**Comment by bseeger** [comment]
@birkland  - I added a comment on that ticket.  No, that ticket doesn't affect export.  Any changes, going forward, to the Repository Item or Collection Item will affect Export.

File: codebase/config/sync/views.view.export_metadata.yml [hunk 1]
```diff
@@ -3475,7 +3475,7 @@ display:
           click_sort_column: uri
           type: link
           settings:
-            trim_length: 80
+            trim_length: 500
```
Comment (bseeger) [inline codebase/config/sync/views.view.export_metadata.yml:3478]: this is arguably outside the scope of the ticket and something I "sneaked" in.  I wanted to change these as I noticed them and not have the overhead of coming back later to fix this separate issue.

Comment (bseeger) [reply codebase/config/sync/views.view.export_metadata.yml:3478]: (this includes the few other spots where I increased the length as well)

**Review by birkland — COMMENTED** [review]
Github UI won't let me approve (probably because I started this PR), but it looks good to me and is approved!

**Pull request merged**